### DISCUSSION TITLE
ERROR HTTP SENDER

### DISCUSSION THREAD

**Discussion by 5011-5011** [nextgenhealthcare/connect#5555] [Q&A]
I made a channel for authentication, it generates the token. You must return it to Postman there I am testing. However, when I create the recipient for sending the token by applying this configuration, that error appears and I cannot continue. I appreciate any help. Thank you

**x-www-form-urlencoded**

![image](https://user-images.githubusercontent.com/98954172/204846291-251f2f14-2bf7-4bc3-87e5-ada9cad1d6d4.png)

**Comment by jonbartels** [comment]
Are there any fields in red?

If you validate your transformer steps, do you get errors about the transformer javascript?

**Reply by pacmano1** [reply]
You need to have at least one query parameter for that content type.

**Reply by 5011-5011** [reply]
@jonbartels 
Transformer is correct. I have no fields in red. When I changed the "CONTENT TYPE" that error message started to appear.

**Reply by jonbartels** [reply]
I reproduced the issue.

Add a value to the "Query parameters" I think that `application/x-www-form-urlencoded` requires at least one query parameter.

**Reply by pacmano1** [reply]
I already said what was wrong 35 minutes ago.  Add a query parameter, even a fake one and the channel will not complain any more.

**Reply by 5011-5011** [reply]
@pacmano1  
Correct, adding a parameter worked perfectly. 
Now I have this error.
![image](https://user-images.githubusercontent.com/98954172/204866631-3b0bcc4e-c146-46bf-a319-6ba307b4723b.png)


In Postman it gives me a 200OK response, however the token does not reach Postman. Any suggestions? Thank you

**Reply by jonbartels** [reply]
Are you making a request from Postman and expecting the token as a response?
Are you using postman as an HTTP listener and making Mirth call postman with a new HTTP request?

**Reply by 5011-5011** [reply]
@jonbartels  
1. From postman I send the request with the parameters in the body.
2. Mirth listens to Postman receives the request, and creates the token using a JavaScript type transformer.
3. The destination configuration is the one I showed a while ago.
I expect Postman to receive the token in conjunction with the response.
I don't know if my flow is correct.

**Comment by jonbartels** [comment]
https://docs.nextgen.com/bundle/Mirth_User_Guide_40/page/connect/connect/topics/c_HTTP_Listener_connect_ug.html

I think you want to:
* change your destination to just a channel writer. The destination transformer will build the auth token and put it into channel map
* add a response header to your source connector so that Mirth responds to Postman with the Authentication token

This sends the token as a REPLY to postman.

**Reply by 5011-5011** [reply]
@jonbartels  Thank you very much for the help! :)
